pandas/core/indexes/multi.py

```diff
@@ -1359,6 +1359,9 @@ def is_monotonic_increasing(self):
         increasing) values.
         """
 
+        if all([x.is_monotonic for x in self.levels]):
```

Comment:
can you add a comment on what this short-circuits. how often is this hit (as compared to the following path)?

Reply:
Of course - it appears to almost always get hit, as it seems levels get sorted as part of `.codes` creation.

Reply:
hmm, when is it *not* hit? 

can you simply sort if its not hit then call the same code (and remove what's below)?

Reply:
Having `.levels` be monotonic gives us the property that monotonicity in code-space is monotonic in level-space (as the values in a level are unique). If we sort a level to obtain monotonicity, we'd have to re-encode that level.

One case is when `MultiIndex(codes=..., levels=...)` is directly constructed and the levels are not sorted. It could be a win to re-encode here, but I'll need to do some testing.

Reply:
ok, not a big deal; the reason is mainly that we can remove code :-> happy to merge this and can investigate that as a followup (or here ok too). lmk.

Reply:
Let's merge this for now - I created a followup issue here: https://github.com/pandas-dev/pandas/issues/27498